Detect new brands: Leff, Tanix and Detect devices for existing brands: Lenovo, Huawei, Meizu, Chuwi, Vivax

sanchezzzhak opened the pull request:
issue #6517

sanchezzzhak committed:
update README.md

sanchezzzhak committed:
Merge pull request #7 from sanchezzzhak/updatereadme

Update README.md

sanchezzzhak committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	README.md

sanchezzzhak committed:
Merge remote-tracking branch 'upstream/master'

sanchezzzhak committed:
Merge remote-tracking branch 'upstream/master'

sanchezzzhak committed:
fix(device) detect common model for brand Huawei

issue #6517

sanchezzzhak committed:
feat(device) detect brand Huawei model: MediaPad T 10
feat(device) detect brand Lenovo models: Tab M10 FHD Plus (TB-X606F), TAB4 8 (TB-8504X),
 Tab E7 (TB-7104I), TAB3 10 Plus (TB3-X70L), TAB3 8 Plus (TB-8703X), TAB2 A10 (A10-70L)

issue #6517

sanchezzzhak committed:
feat(device) detect brand Lenovo models: Tab M10 FHD Plus (TB-X606X), TAB 2 A7 (A7-30HC, A7-30GC),
 TAB4 8 Plus (TB-8704X), Tab 7 Essential (TB-7304X), Tab 7 (TB-7504X)
feat(device) detect brand Meizu model: 15 Lite

issue #6517

sanchezzzhak committed:
feat(device) detect new brand BOE model: HV320WHB-N55
feat(device) detect brand Rombica model: Smart Box v009

issue #6517

sanchezzzhak committed:
feat(device) rename new brand BOE to Leff model: HV320WHB-N55

issue #6517

sanchezzzhak committed:
fix: tests

sanchezzzhak committed:
feat(device) detect brand Huawei model: Honor Pad V6 (KRJ-W09)
feat(device) detect brand Lenovo models: TAB3 7 Plus (TB-7703X), M10 (TB-X505X), Tab M8 (TB-8505F)

issue #6517

sanchezzzhak committed:
feat(device) detect brand Lenovo model: TAB 2 A7 (A7-30DC)
feat(device) detect brand Chuwi model: Hi9 Air
feat(device) rename model Lenovo Tab 4 10 Plus => Lenovo TAB4 10 Plus

issue #6517

sanchezzzhak commented on Tests/fixtures/smartphone-19.yml:
```diff
@@ -557,3 +557,399 @@
+  os_family: Android
+  browser_family: Chrome
+-
+  user_agent: Mozilla/5.0 (Linux; Android 4.4.4; Leff Shell HV320WHB-N55) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.102 YaBrowser/18.11.1.1011.01 Safari/537.36
+  os:
+    name: Android
+    version: 4.4.4
+    platform: ""
+  client:
+    type: browser
+    name: Yandex Browser
+    version: 18.11.1.1011.01
+    engine: Blink
+    engine_version: ""
+  device:
+    type: tv
+    brand: Leff
+    model: HV320WHB-N55
+  os_family: Android
+  browser_family: Unknown
```
@sgiehl 

This definition bothers me.

Leff Shell HV320WHB-N55
Shell Prefix-means that the mainboard or panel number goes next
HV320WHB-N55  - Mainboard ID from the manufacturer BOE
![image](https://user-images.githubusercontent.com/1337066/101754134-74b67080-3ae4-11eb-86b8-4ee4c7654adb.png)

a similar mainboard is used in the brand Manta model 32lha19s, Digma DM-LED32R301BT2S etc

Leave the model empty?

sanchezzzhak commented:
@sgiehl 

This definition bothers me.

Leff Shell HV320WHB-N55
Shell Prefix-means that the mainboard or panel number goes next
HV320WHB-N55  - Mainboard ID from the manufacturer BOE
![image](https://user-images.githubusercontent.com/1337066/101754134-74b67080-3ae4-11eb-86b8-4ee4c7654adb.png)
https://tvscreen.en.made-in-china.com/product/AOynpreJJUhW/China-Good-TV-Panel-Module-HD-32-Inch-TV-Open-Cell-Hv320whb-N55.html
a similar mainboard is used in the brand Manta model 32lha19s, Digma DM-LED32R301BT2S etc

Leave the model empty?

sgiehl commented:
@sanchezzzhak 
If it doesn't point to an explicit device model, guess it makes sense to leave it empty. 
Or if it is never possible for that brand maybe some generic model like `Generic Smart TV` or similar might make sense

sanchezzzhak committed:
fix brand Leff

sanchezzzhak committed:
feat(device) detect new brand Tanix model: TX5 Max
feat(device) detect brand Lenovo model: Yoga Tab 3 Plus (YT-X703L)
feat(device) detect brand Vivax model: Fly V1

issue #6517

sanchezzzhak committed:
feat(test) added test for brand Tanix model: TX5 Max

issue #6517

sgiehl committed:
Merge branch 'master' into 6517
